src/main/java/io/confluent/kafkarest/SimpleConsumerRecordsCache.java

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright 2016 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.kafkarest;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.TopicPartition;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NavigableMap;
+import java.util.Queue;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+
+
+
+public class SimpleConsumerRecordsCache {
```

Comment:
I'm curious about the need for a cache here. What's the usage pattern that would make this effective? I'm curious because adding a cache necessarily adds configs (e.g. you at least need to be able to specify the size) and I'm not sure that consumers in general should expect to benefit from caching the data they read.

Is there a use case you're thinking of specifically which would benefit from this caching? Of course, generically there is any case where there are multiple readers of the same data, but should we expect that to be especially common? Perhaps a bit more common with simple consumers than with consumer groups, but enough to warrant a cache?